s3store/s3store.go

```diff
@@ -547,3 +561,28 @@ func isAwsError(err error, code string) bool {
 	}
 	return false
 }
+
+func (store S3Store) CalcOptimalPartSize(size *int64) int64 {
```

Comment:
> Therefore I was trying to be as efficient on memory-use as possible, as these are quite a lot of test cases.

Please don't do this kind of premature optimization. Unless you have proven using a benchmark that this change provides an improvement, stay away for that. Code readability and ease to understanding it is usually more important than performance.

Also I have serious doubt that this would bring any speed-ups since you need to dereference the pointer multiple times and also need to allocate memory for the pointer which could mean even worse memory-efficient. However, I am not a 100% sure on how Golang handles stack pointers but it doesn't matter here.

> Do you want me to change that?

Yes, please do so.